code/game/objects/structures/watercloset.dm

```diff
@@ -480,6 +459,151 @@
 		return
 	return ..()
 
+//Water source, use the type water_source for unlimited water sources like classic sinks.
+/obj/structure/water_source
+	name = "Water Source"
+	icon = 'icons/obj/watercloset.dmi'
+	icon_state = "sink"
+	desc = "A sink used for washing one's hands and face. This one seems to be infinite!"
+	anchored = TRUE
+	var/busy = FALSE 	//Something's being washed at the moment
+	var/dispensedreagent = /datum/reagent/water // for whenever plumbing happens
+
+/obj/structure/water_source/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+	if(!user || !istype(user))
+		return
+	if(!iscarbon(user))
+		return
```

Comment:
```suggestion
	if(!iscarbon(user))
		return
```
`iscarbon` will filter the above two conditions already

Reply:
Huh, didn't know that!